cmd/ncproxy/hcn.go

```diff
@@ -5,41 +5,56 @@ package main
 import (
 	"context"
 	"encoding/json"
+	"net"
+	"strings"
 
 	"github.com/Microsoft/hcsshim/hcn"
 	ncproxygrpc "github.com/Microsoft/hcsshim/pkg/ncproxy/ncproxygrpc/v1"
 	"github.com/pkg/errors"
 )
 
 func hcnEndpointToEndpointResponse(ep *hcn.HostComputeEndpoint) (_ *ncproxygrpc.GetEndpointResponse, err error) {
+	hcnEndpointResp := &ncproxygrpc.HcnEndpointSettings{
+		Name:        ep.Name,
+		Macaddress:  ep.MacAddress,
+		NetworkName: ep.HostComputeNetwork,
+		DnsSetting: &ncproxygrpc.DnsSetting{
+			ServerIpAddrs: ep.Dns.ServerList,
+			Domain:        ep.Dns.Domain,
+			Search:        ep.Dns.Search,
+		},
+	}
+
 	policies, err := parseEndpointPolicies(ep.Policies)
 	if err != nil {
 		return nil, err
 	}
-	ipConfigInfo := ep.IpConfigurations
-	if len(ipConfigInfo) == 0 {
-		return nil, errors.Errorf("failed to find network %v ip configuration information", ep.Name)
+	hcnEndpointResp.Policies = policies
+
+	ipConfigInfos := ep.IpConfigurations
+	// there may be one ipv4 and/or one ipv6 configuration for an endpoint
+	if len(ipConfigInfos) == 0 || len(ipConfigInfos) > 2 {
+		return nil, errors.Errorf("invalid number (%v) of ip configuration information for endpoint %v", len(ipConfigInfos), ep.Name)
+	}
+	for _, ipConfig := range ipConfigInfos {
+		ip := net.ParseIP(ipConfig.IpAddress)
+		if ip.To4() != nil {
```

Comment:
Nit: I'd hope the endpoint would never have something that doesn't parse as an IP, but ParseIP returns nil if it's not a valid textual representation of an IP, so it's possible we could nil deref here.

Reply:
Oh never mind! It's a byte slice under the hood, so we're good